frontend/src/lib/k8s/hpa.ts

```diff
@@ -0,0 +1,366 @@
+                  status && status.resource
+                    ? metricValueStatus(status.resource.current, t)
+                    : t('frequent|<unknown>')
+                }/${metricTargetValue(spec.resource.target)}`;
+              }
+              if (spec.resource.target.type === 'Utilization') {
+                definition = `${definition} ${defineMetricTarget(spec.resource.target)}`;
+                if (status) {
+                  value = `${
+                    status.resource
+                      ? status.resource.current.averageUtilization
+                      : t('frequent|<unknown>')
+                  }% (${
+                    status.resource ? status.resource.current.averageValue : t('frequent|<unknown>')
+                  })/${metricTargetValue(spec.resource.target)}`;
+                  shortValue = `${
+                    status.resource
+                      ? status.resource.current.averageUtilization
+                      : t('frequent|<unknown>')
+                  }% /${metricTargetValue(spec.resource.target)}`;
+                } else {
+                  value = `${t('frequent|<unknown>')}/${metricTargetValue(spec.resource.target)}`;
+                  shortValue = `${t('frequent|<unknown>')}/${metricTargetValue(
+                    spec.resource.target
+                  )}`;
+                }
+              }
+              metrics.push({
+                name: spec.resource.name,
+                definition,
+                value,
+                shortValue: shortValue,
+              });
+            }
+          }
+          break;
+        case 'ContainerResource':
+          {
+            if (spec.containerResource) {
+              const value = `${
+                status && status.containerResource
+                  ? metricValueStatus(status.containerResource.current, t)
+                  : t('frequent|<unknown>')
+              }/${metricTargetValue(spec.containerResource.target)}`;
+              const definition = t(
+                'hpa|resource {{resourceName}} of container {{containerName}} on pods',
+                {
+                  resourceName: spec.containerResource.name,
+                  containerName: spec.containerResource.container,
+                }
+              );
+              metrics.push({
+                name: spec.containerResource.name,
+                definition,
+                value,
+                shortValue: value,
+              });
+            }
+          }
+          break;
+      }
+    }
+    return metrics;
+  }
+
+  static async getEvents(namespace: string, name: string) {
+    const path = `/api/v1/namespaces/${namespace}/events`;
+    const params: QueryParameters = {
+      fieldSelector: `involvedObject.name=${name},involvedObject.kind=HorizontalPodAutoscaler,involvedObject.namespace=${namespace}`,
+      limit: 500,
+    };
+    const response = await request(path, {}, true, true, params);
+    return response.items;
+  }
```

Comment:
This is a generic events fetcher function, so AFAIU, we should define it outside of the HPA class.
A probably better idea is to use it right from the event, like: `const events = Event.useList({namespace, name})` which would fetch it as regularly but using the query params, and making it a hook (be careful with likely wanting to call it conditionally, on the existence of namespace/name, as that logic needs to be moved into the hook).

Reply:
Moved this to a separate component.